### PR TITLE
DO-246: Progress bar added.

### PR DESCRIPTION
**Pull request by amitjain24524**

### CODE REVIEW

File: src/utils/common/formatters/formmatters.js [hunk 1]
```diff
@@ -85,3 +85,27 @@ export const removeLastStringContactDetails = (item) => {
   }
   return contactDetails;
 };
+
+export const setProgressBarPathColor = ((progress) => {
+  let pathColor = '';
+  if (progress < '33.33') {
+    pathColor = '#cf3942';
```
Comment (pepcus-ankit) [inline src/utils/common/formatters/formmatters.js:92]: Constant

File: src/utils/common/formatters/formmatters.js [hunk 2]
```diff
@@ -85,3 +85,27 @@ export const removeLastStringContactDetails = (item) => {
   }
   return contactDetails;
 };
+
+export const setProgressBarPathColor = ((progress) => {
+  let pathColor = '';
+  if (progress < '33.33') {
```
Comment (pepcus-ankit) [inline src/utils/common/formatters/formmatters.js:91]: Constant

File: src/utils/common/formatters/formmatters.js [hunk 3]
```diff
@@ -85,3 +85,27 @@ export const removeLastStringContactDetails = (item) => {
...
+    pathColor = '#cf3942';
+  } else if (progress < '66.66') {
+    pathColor = '#ffcf1a';
+  } else {
+    pathColor = '#1aa75d';
+  }
+  return pathColor;
+});
+
+export const setProgressBarTextColor = ((progress) => {
```
Comment (pepcus-ankit) [inline src/utils/common/formatters/formmatters.js:101]: Method Comment

File: src/utils/common/formatters/formmatters.js [hunk 4]
```diff
@@ -85,3 +85,27 @@ export const removeLastStringContactDetails = (item) => {
   }
   return contactDetails;
 };
+
+export const setProgressBarPathColor = ((progress) => {
```
Comment (pepcus-ankit) [inline src/utils/common/formatters/formmatters.js:89]: Method comment

File: src/components/common/ProgressBar/ProgressBar.jsx [hunk 5]
```diff
@@ -1,25 +1,51 @@
...
-  const { progress } = props;
+  const { progress, profileImg, onClickImage } = props;
+  const pathColorStyle = setProgressBarPathColor(progress);
+  const textColorStyle = setProgressBarTextColor(progress);
   return (
-    <div className="progress-bar">
-      <div className="progress" style={{ width: `${progress}%` }}>
-        <span>{progress}%</span>
-      </div>
+    <div className="progress-container">
```
Comment (mohitsaladiya) [inline src/components/common/ProgressBar/ProgressBar.jsx:13]: Use container component
```suggestion
    <Container className="progress-bar__container">
```

File: src/components/routes/Member/AuthenticatedMember/MemberDetailsViewRoute/MemberDetailsViewRoute.jsx [hunk 6]
```diff
@@ -113,13 +113,11 @@ class MemberDetails extends React.Component {
...
                     <div className="details__row main-detail-section">
                       <div className="details__col-50">
-                        <figure className="mbr-image">
-                          <img
-                            src={memberDetails.imageUrl}
-                            alt="profileImg"
-                            onClick={this.handleOnProfileImgClick}
-                          />
-                        </figure>
+                        <ProgressBar
```
Comment (mohitsaladiya) [inline src/components/routes/Member/AuthenticatedMember/MemberDetailsViewRoute/MemberDetailsViewRoute.jsx:116]:
```suggestion
                     <ProgressBar
                          progress={memberDetails.memberDetailsScore}
                        >
                         <img
                            src={memberDetails.imageUrl}
                            alt="profileImg"
                            onClick={this.handleOnProfileImgClick}
                          />
                       </ProgressBar>
                        

                    
```

File: src/components/routes/Member/AuthenticatedMember/MemberProfileViewRoute/MemberProfileEditView/MemberProfileEditView.jsx [hunk 7]
```diff
@@ -143,14 +143,11 @@ class MemberProfileEditView extends React.Component {
...
             <div className="mbr-detail-edit__image-container">
-              <figure className="mbr-image">
-                <img
-                  className="profile-image"
-                  src={memberDetails.imageUrl}
-                  alt="profileImg"
-                  onClick={this.handleOnProfileImgClick}
-                />
-              </figure>
+              <ProgressBar
```
Comment (mohitsaladiya) [inline src/components/routes/Member/AuthenticatedMember/MemberProfileViewRoute/MemberProfileEditView/MemberProfileEditView.jsx:146]:
```suggestion
                     <ProgressBar
                          progress={memberDetails.memberDetailsScore}
                        >
                         <img
                            src={memberDetails.imageUrl}
                            alt="profileImg"
                            onClick={this.handleOnProfileImgClick}
                          />
                       </ProgressBar>
```

File: src/components/routes/Member/AuthenticatedMember/MemberProfileViewRoute/MemberProfileView/MemberProfileView.jsx [hunk 8]
```diff
@@ -132,13 +133,11 @@ class MemberProfileView extends React.Component {
...
                       <div className="details__row main-detail-section">
                         <div className="details__col-50">
-                          <figure className="mbr-image">
-                            <img
-                              src={memberDetails.imageUrl}
-                              alt="profileImg"
-                              onClick={this.handleOnProfileImgClick}
-                            />
-                          </figure>
+                          <ProgressBar
```
Comment (mohitsaladiya) [inline src/components/routes/Member/AuthenticatedMember/MemberProfileViewRoute/MemberProfileView/MemberProfileView.jsx:136]:
```suggestion
                          <ProgressBar
                          progress={memberDetails.memberDetailsScore}
                        >
                         <img
                            src={memberDetails.imageUrl}
                            alt="profileImg"
                            onClick={this.handleOnProfileImgClick}
                          />
                       </ProgressBar>
```

File: src/utils/common/formatters/formmatters.js [hunk 9]
```diff
@@ -85,3 +85,27 @@ export const removeLastStringContactDetails = (item) => {
...
+  } else {
+    pathColor = '#1aa75d';
+  }
+  return pathColor;
+});
+
+export const setProgressBarTextColor = ((progress) => {
+  let textColor = '';
+  if (progress < '33.33') {
+    textColor = '#fff';
```
Comment (mohitsaladiya) [inline src/utils/common/formatters/formmatters.js:104]: Create constant for progress bar text colors in common constants.
 PROGRESS_BAR_TEXT_COLORS = {
  LOW: '#fff',
  HIGH: '#312a2a',
}

File: src/utils/common/formatters/formmatters.js (hunk 3 above)
Comment (mohitsaladiya) [reply src/utils/common/formatters/formmatters.js:101]: Write JSDoc for the method

File: src/utils/common/formatters/formmatters.js [hunk 10]
```diff
@@ -85,3 +85,27 @@ export const removeLastStringContactDetails = (item) => {
   }
   return contactDetails;
 };
+
+export const setProgressBarPathColor = ((progress) => {
+  let pathColor = '';
+  if (progress < '33.33') {
+    pathColor = '#cf3942';
+  } else if (progress < '66.66') {
+    pathColor = '#ffcf1a';
```
Comment (mohitsaladiya) [inline src/utils/common/formatters/formmatters.js:94]: PROGRESS_BAR_COLORS = {
  LOW: '#xxxx',
  HIGH: '#xxxxxx',
}

File: src/utils/common/formatters/formmatters.js (hunk 4 above)
Comment (mohitsaladiya) [reply src/utils/common/formatters/formmatters.js:89]: Write JsDoc

File: src/components/common/ProgressBar/ProgressBar.jsx [hunk 11]
```diff
@@ -1,25 +1,51 @@
...
 import PropTypes from 'prop-types';
-
+import { buildStyles, CircularProgressbarWithChildren } from 'react-circular-progressbar';
+import 'react-circular-progressbar/dist/styles.css';
+import { setProgressBarPathColor, setProgressBarTextColor } from '../../../utils/common/formatters/formmatters';
 import './_ProgressBar.scss';
 
 const ProgressBar = (props) => {
-  const { progress } = props;
+  const { progress, profileImg, onClickImage } = props;
```
Comment (mohitsaladiya) [inline src/components/common/ProgressBar/ProgressBar.jsx:9]: We can not have props specific to image in Progress Bar.
```suggestion
  const { progress } = props;
```

File: src/utils/common/formatters/formmatters.js (hunk 1 above)
Comment (amitjain24524) [reply src/utils/common/formatters/formmatters.js]: done

File: src/utils/common/formatters/formmatters.js (hunk 2 above)
Comment (amitjain24524) [reply src/utils/common/formatters/formmatters.js]: done

File: src/utils/common/formatters/formmatters.js (hunk 3 above)
Comment (amitjain24524) [reply src/utils/common/formatters/formmatters.js:105]: done

File: src/utils/common/formatters/formmatters.js (hunk 4 above)
Comment (amitjain24524) [reply src/utils/common/formatters/formmatters.js:92]: done

File: src/components/common/ProgressBar/ProgressBar.jsx [hunk 12]
```diff
@@ -1,25 +1,51 @@
...
+        circleRatio={0.75}
+        styles={buildStyles({
+                rotation: 1 / 2 + 1 / 8,
+                trailColor: "#ccc9c9",
+                pathColor: pathColorStyle,
+                pathTransitionDuration: 2
+        })}
+      >
+        {/* Put any JSX content in here that you'd like. It'll be vertically and horizonally centered. */}
+        <img
```
[the remark this answers is not in the document]
Comment (amitjain24524) [reply src/components/common/ProgressBar/ProgressBar.jsx]: done

File: src/components/common/ProgressBar/ProgressBar.jsx (hunk 5 above)
Comment (amitjain24524) [reply src/components/common/ProgressBar/ProgressBar.jsx]: done

File: src/components/common/ProgressBar/ProgressBar.jsx (hunk 11 above)
Comment (amitjain24524) [reply src/components/common/ProgressBar/ProgressBar.jsx]: done

File: src/utils/common/formatters/formmatters.js (hunk 10 above)
Comment (amitjain24524) [reply src/utils/common/formatters/formmatters.js]: done

File: src/utils/common/formatters/formmatters.js (hunk 9 above)
Comment (amitjain24524) [reply src/utils/common/formatters/formmatters.js]: done

File: src/components/routes/Member/AuthenticatedMember/MemberProfileViewRoute/MemberProfileView/MemberProfileView.jsx (hunk 8 above)
Comment (amitjain24524) [reply src/components/routes/Member/AuthenticatedMember/MemberProfileViewRoute/MemberProfileView/MemberProfileView.jsx:136]: done

File: src/components/routes/Member/AuthenticatedMember/MemberProfileViewRoute/MemberProfileEditView/MemberProfileEditView.jsx (hunk 7 above)
Comment (amitjain24524) [reply src/components/routes/Member/AuthenticatedMember/MemberProfileViewRoute/MemberProfileEditView/MemberProfileEditView.jsx:146]: done

File: src/components/routes/Member/AuthenticatedMember/MemberDetailsViewRoute/MemberDetailsViewRoute.jsx (hunk 6 above)
Comment (amitjain24524) [reply src/components/routes/Member/AuthenticatedMember/MemberDetailsViewRoute/MemberDetailsViewRoute.jsx:116]: done

**Pull request merged**